### PR TITLE
Fixes #442: Added `print.h2o4gpu_model()` to cleanup model object output

### LINKED ISSUE #442: Add `print` method to `h2o4gpu_model` object
Right now, if we print the model object, the output is messy and ugly. We need to clean it up.

### PR DESCRIPTION
**Pull request by terrytangyuan**
* Added `print.h2o4gpu_model()`
* Added `description` field for each model

Sample output of `print(model)`:

```
Algorithm: KMeans Clustering

Parameters:
algorithm = "auto", copy_x = TRUE, init = "k-means++", max_iter = 300, n_clusters = 2, n_init = 1, n_jobs = 1, precompute_distances = "auto", random_state = 1234, tol = 1e-04, verbose = 0
```

### CODE REVIEW

File: src/interface_r/R/auto_generated_wrappers.R [hunk 1]
```diff
@@ -465,6 +465,6 @@ h2o4gpu.truncated_svd <- function(
     n_gpus = as.integer(n_gpus),
     gpu_id = as.integer(gpu_id)
   )
-  h2o4gpu_model(model, NULL)
+  h2o4gpu_model(model, NULL, "Truncated Singular-value Decomposition")
```
Comment (navdeep-G) [inline src/interface_r/R/auto_generated_wrappers.R]: Minor: Please change to `Truncated Singular Value Decomposition (TruncatedSVD)`

File: src/interface_r/R/auto_generated_wrappers.R [hunk 2]
```diff
@@ -439,7 +439,7 @@ h2o4gpu.pca <- function(
     verbose = verbose,
     backend = backend
   )
-  h2o4gpu_model(model, NULL)
+  h2o4gpu_model(model, NULL, "Principal Components Analysis")
```
Comment (navdeep-G) [inline src/interface_r/R/auto_generated_wrappers.R]: Minor: Please change to `Principal Components Analysis (PCA)`

**Comment by terrytangyuan** [comment]
@navdeep-G Done!

**Comment by navdeep-G** [comment]
@ledell Can you please have a quick look?

**Pull request merged**